app/src/main/kotlin/com/wire/android/ui/WireActivity.kt

```diff
@@ -374,19 +376,19 @@ class WireActivity : AppCompatActivity() {
                     onConfirm = {
                         featureFlagNotificationViewModel.dismissTeamAppLockDialog()
                         if (isTeamAppLockEnabled) {
-                            val isUserAppLockSet =
-                                featureFlagNotificationViewModel.isUserAppLockSet()
-                            // No need to setup another app lock if the user already has one
-                            if (!isUserAppLockSet) {
-                                Intent(this@WireActivity, AppLockActivity::class.java)
-                                    .apply {
-                                        putExtra(AppLockActivity.SET_TEAM_APP_LOCK, true)
-                                    }.also {
-                                        startActivity(it)
+                            featureFlagNotificationViewModel.isUserAppLockSet()
+                                .flowWithLifecycle(lifecycle, Lifecycle.State.STARTED)
+                                .onEach { isUserAppLockSet ->
+                                    // No need to setup another app lock if the user already has one
+                                    when (isUserAppLockSet) {
+                                        true -> featureFlagNotificationViewModel.markTeamAppLockStatusAsNot()
+                                        false -> startActivity(
```

Comment:
So `AppLockActivity` will be started every time, when `featureFlagNotificationViewModel.isUserAppLockSet()` flow emits `true`. 
So for example: 
- user has not locked app
-  go to settings
- set AppLock to true

Result: app locked. 
But that is not what we really want 😅

Reply:
@borichellow Not exactly, the mentioned scenario does not happen, because the flow will only emit a value if the new one is different from the last one.
[Screen_recording_20241119_114719.webm](https://github.com/user-attachments/assets/14f6f24e-ed59-4359-8890-029cb12c190e)

Reply:
@borichellow Stupidiest stuff i have ever made 👎🏻 . Absolutely right, I skipped many stuff. Sorry about that i am trying to grasp all the underlying stuff and I am new to this. Ty for review, I am closing this PR